sui_programmability/adapter/src/adapter.rs

```diff
@@ -625,34 +585,40 @@ fn handle_transfer<
 }
 
 #[cfg(debug_assertions)]
-fn check_transferred_object_invariants(new_object: &MoveObject, old_object: &Option<Object>) {
-    if let Some(o) = old_object {
+fn check_transferred_object_invariants(
+    new_object: &MoveObject,
+    old_object: &Option<(object::Owner, SequenceNumber)>,
+) {
+    if let Some((_owner, old_version)) = old_object {
         // check consistency between the transferred object `new_object` and the tx input `o`
         // specifically, the object id, type, and version should be unchanged
-        let m = o.data.try_as_move().unwrap();
-        debug_assert_eq!(m.id(), new_object.id());
-        debug_assert_eq!(m.version(), new_object.version());
-        debug_assert_eq!(m.type_, new_object.type_);
+        // we can only check the version here
+        debug_assert_eq!(*old_version, new_object.version());
     }
 }
 
 pub struct TypeCheckSuccess {
     pub module_id: ModuleId,
+    // object_owner_map maps from object ID to its exclusive object owner.
+    // This map will be used for detecting circular ownership among
+    // objects, which can only happen to objects exclusively owned
+    // by objects.
```

Comment:
These comments belong to where we were creating `object_owner_map` from object_data